sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubApplicationInsightsTests.cs

```diff
@@ -0,0 +1,430 @@
+
+            Assert.AreEqual(operationName, request.Context.Operation.Name);
+            Assert.AreEqual(operationName, request.Name);
+
+            Assert.True(request.Properties.ContainsKey(LogConstants.InvocationIdKey));
+
+            // EventHub does not populate Trigger reason
+            Assert.False(request.Properties.ContainsKey(LogConstants.TriggerReasonKey));
+
+            StringAssert.StartsWith("webjobs:", request.Context.GetInternalContext().SdkVersion);
+
+            Assert.AreEqual(success, request.Success);
+            Assert.AreEqual("0", request.ResponseCode);
+
+            Assert.False(request.Properties.Any(p => p.Key == LogConstants.SucceededKey));
+        }
+
+        private void ValidateEventHubDependency(
+            DependencyTelemetry dependency,
+            string endpoint,
+            string entityName,
+            string name,
+            string operationName,
+            string operationId,
+            string parentId,
+            string category)
+        {
+            Assert.AreEqual($"{endpoint}/{entityName}", dependency.Target);
+            Assert.AreEqual("Azure Event Hubs", dependency.Type);
+            Assert.AreEqual(name, dependency.Name);
+            Assert.True(dependency.Success);
+            Assert.True(string.IsNullOrEmpty(dependency.Data));
+
+            Assert.AreEqual(category, dependency.Properties[LogConstants.CategoryNameKey]);
+            Assert.AreEqual(LogLevel.Information.ToString(), dependency.Properties[LogConstants.LogLevelKey]);
+            Assert.NotNull(dependency.Target);
+            Assert.NotNull(dependency.Name);
+            Assert.NotNull(dependency.Id);
+
+            if (operationId != null)
+            {
+                Assert.AreEqual(operationId, dependency.Context.Operation.Id);
+            }
+
+            Assert.AreEqual(operationName, dependency.Context.Operation.Name);
+
+            if (parentId != null)
+            {
+                Assert.AreEqual(parentId, dependency.Context.Operation.ParentId);
+            }
+
+            Assert.True(dependency.Properties.ContainsKey(LogConstants.InvocationIdKey));
+        }
+
+        public class EventHubTestSingleDispatchJobs
+        {
+            public static void SendEvent_TestHub(string input, [EventHub(TestHubName)] out EventData evt)
+            {
+                evt = new EventData(Encoding.UTF8.GetBytes(input));
+            }
+
+            public static void ProcessSingleEvent([EventHubTrigger(TestHubName)] string evt)
+            {
+                if (evt.StartsWith(_testId))
+                {
+                    _eventWait.Set();
+                }
+            }
+        }
+
+        public class EventHubTestMultipleDispatchJobs
+        {
+            public static List<int> LinksCount = new List<int>();
+            public static int MessagesCount = 0;
+
+            private const int EventCount = 5;
+            private static readonly object MessageLock = new object();
```

Comment:
These tests a bit static-happy but I'm not exactly sure how to get this information out otherwise.